src/openms/source/ANALYSIS/OPENSWATH/MRMAssay.cpp

```diff
@@ -159,22 +159,46 @@ namespace OpenMS
   {
     std::vector<OpenMS::AASequence> modified_sequences;
     bool multi_mod_switch = false;
+    bool skip_invalid_mod_seq = false;
 
     for (std::vector<OpenMS::AASequence>::const_iterator sq_it = sequences.begin(); sq_it != sequences.end(); ++sq_it)
     {
       for (std::vector<std::vector<size_t> >::const_iterator mc_it = mods_combs.begin(); mc_it != mods_combs.end(); ++mc_it)
       {
         multi_mod_switch = false;
+        skip_invalid_mod_seq = false;
         OpenMS::AASequence temp_sequence = *sq_it;
         for (std::vector<size_t>::const_iterator pos_it = mc_it->begin(); pos_it != mc_it->end(); ++pos_it)
         {
           if (*pos_it == 0)
           {
-            temp_sequence.setNTerminalModification(modification);
+            // Check first to make sure ending residue is NTerm modifiable
+            const ResidueModification* modifiable_any_nterm = ModificationsDB::getInstance()->getModification(modification,"X",ResidueModification::N_TERM);
+            const ResidueModification* modifiable_nterm = ModificationsDB::getInstance()->getModification(modification,"",ResidueModification::N_TERM);
+            if ( temp_sequence[0].getOneLetterCode() == OpenMS::String(modifiable_nterm->getOrigin()) || OpenMS::String(modifiable_any_nterm->getOrigin()) == "X" ) {
```

Comment:
converting a char to string is not needed here. Just compare the chars (using `'X'` for the second comparison instead of `"X"`)